### DISCUSSION TITLE
Ask any questions you like!

### DISCUSSION THREAD

**Discussion by spamegg1** [spamegg1/Epp-Discrete-Math-5th-solutions#1] [Q&A]
Welcome! :tada:

**Comment by ahmedhalimx** [comment]
I think there is a mistake In Chapter 3, Exersice Set 3.2, Question 34 (a).
PDF solution: If n _is not divisible_ by any prime number from 2 through √n, then n is not prime.
My Solution: If n _is divisible_ by any prime number from 2 through √n, then n is not prime.
Which is right?

**Reply by spamegg1** [reply]
You are correct; this was fixed [just recently](https://github.com/spamegg1/Epp-Discrete-Math-5th-solutions/pull/24/files) so you might want to get the latest release.

**Comment by cything** [comment]
Is 3.3.41(g) question incorrect?

**Reply by spamegg1** [reply]
I don't think so, why is it incorrect? Can you explain?

**Reply by cything** [reply]
You have it as $\exists x \in \mathbb{R}$ such that $\forall y \in \mathbb{R}$, $y < x$ (Epp.tex#9181). In the book the question is $\forall x \in \mathbb{R}^+, \exists y \in \mathbb{R}^+$ such that $y<x$.

**Comment by elfaidi-ghassen** [comment]
I'm having a hard time understanding 4.7.16, can you help me?
in the solution, it says $b > a$ because $b^2 - a^2 = 4 > 0$, I don't understand why that's true in this exercise, I feel I'm missing something. Also, the unique prime factors of 4 are $2 . 2$ and $4 . 1$
but, don't we need to also verify the cases? like $(b + a) = -2$ and $(b - a) = -2$?

**Reply by spamegg1** [reply]
Yes you are right, it would only give us $|b| > |a|$.
I must have implicitly assumed that $a,b$ are positive.

We can actually assume $a,b$ are positive without loss of generality,
because $b^2 = (-b)^2$ and $a^2 = (-a)^2$ so if they are negative,
they can be replaced by $-b, -a$ without affecting anything.

I'll add the "WLOG" to the proof, thanks!

**Comment by mohammadazeemwani** [comment]
In 2.1.46 (c), 
You have shown proving equivalence by truth table. 
I went by simplifying statement forms. 

I have two queries: 
1. When to choose which method to prove logical equivalence. Or should we do simplification first then use truth table (on the simplified statement form)

2. When we use de-morgans law, do we need to keep the bracket? Which one of the the following is correct on basis of that:
<img width="400" height="333" alt="image" src="https://github.com/user-attachments/assets/ef238515-9e6f-4e98-9a92-c986bdb07b9c" />

**Reply by spamegg1** [reply]
Right, the problem does not state which method to use, so you can use any method you want. It said to use truth tables for 40-43, so I guess my mind was still on "truth table mode". :smile: 

1. Yes you can simplify then truth table, that's great. Often these cannot really be simplified, though. Also you have to think about "the purpose of the exercise". Sometimes simplifying can make the exercise a bit pointless, whereas the truth table can be more informative / educational. Or vice versa! Don't think of it in terms of "the most optimal way". We'll start taking short cuts later on, it's best to take the time to build the hard slow logic in early chapters.
2. You can drop brackets as long as it remains correct. You need to remember the precedence of the logical operators. And applies before Or. Not applies before And. So I'd keep brackets at first, then make sure that it's safe to remove. Also sometimes keeping the brackets can be better for readability, it's easier visually (as long as there aren't too many nested brackets).

**Reply by mohammadazeemwani** [reply]
Got it.
Thanks a lot.

**Reply by mohammadazeemwani** [reply]
On the precedence, you mentioned "And applies before Or" but AND and OR are coequal.
<img width="500" alt="image" src="https://github.com/user-attachments/assets/2138baf0-d895-4019-b7a6-74365fce4d6d" />

**Reply by spamegg1** [reply]
That explanation is missing a bit. It says "when both are present, parentheses might be needed" without getting into the details.

$p \wedge q \vee r$ is actually $(p \wedge q) \vee r$ but

$p \vee q \wedge r$ is $p \vee (q \wedge r)$, it's not $(p \vee q) \wedge r$.

**Reply by mohammadazeemwani** [reply]
ooo. Thanks a lot for that.

**Comment by mohammadazeemwani** [comment]
$\text{In 2.2.44}$

<img width="600" alt="image" src="https://github.com/user-attachments/assets/ee0f6832-7f10-4daa-b9d1-26af0b79119e" />

In the then clause, solution says $\text{``it will win the championship''}$

I believe it should be $\text{``they have won the championship''}$

My believe comes from the fact that "the sufficient condition ensures the truth of whether they have won the championship".

Am I missing something??

**Reply by spamegg1** [reply]
Yes there are many ways this can be expressed in English, sometimes both past, present and future tenses can be used in the same place.

It's probably best to stick to neutral present tense, so "then it wins the championship" is a good one.

**Reply by mohammadazeemwani** [reply]
Noted

**Comment by mohammadazeemwani** [comment]
In 2.3.10

<img width="600" alt="file-022e6e40fe8e2db3a3bbdbfee9ecc588" src="https://github.com/user-attachments/assets/5b271a84-21a9-412c-b812-6e7d75059bcc" />

How is this "modes tollens"?
There is no statement saying $\text{not r}$

**Reply by spamegg1** [reply]
Ah you're right, I meant to say "contrapositive".

**Reply by mohammadazeemwani** [reply]
Ok, kindly merge it then #111

**Comment by mohammadazeemwani** [comment]
In 2.3.38 (d)

<img width="700" height="809" alt="image" src="https://github.com/user-attachments/assets/7194b1e1-888e-4624-8795-c2876b204851" />

Your claim that V is knave, should mean "number of knights is atmost 2". But you are saying "atmost 4"?
I can't make sense of it..

Rather I can't make the sense of the approach you followed. It looks like we are (in this solution) using a mixture of writing logical deductions at high level and processing all low-level information in our head. 
Which is unusual to the pattern of reaching conclusions from just logical deductions.
What is your take on this?

**Reply by spamegg1** [reply]
It goes through alphabetically checking the possibilities of each letter being a knight.

Assume U is a knight, follow the logical implications...

Assume V is a knight, follow the logical implications...

etc.

**Reply by spamegg1** [reply]
I'll make that clearer in the text.

**Reply by spamegg1** [reply]
I'll add "by (1), ..." etc. type of explanations too.

**Comment by mohammadazeemwani** [comment]
In 2.4.24
<img width="600" alt="image" src="https://github.com/user-attachments/assets/388819b8-a439-4c4f-b1a2-4f12cb56593d" />

In the solution you say (or assume?) when both the switches are up ($P=1 \text{ and } Q=1$), the lights are off ($R=0$) but nowhere in the problem is it stated such.   

How did you concluded that? 
Am I missing anything?

**Reply by spamegg1** [reply]
In the problem it says

> Assume the lights have been installed so that when both switches are in the down position, the lights are off.

**Reply by mohammadazeemwani** [reply]
Yes it says that and hence the case $P=Q=0 \text{ and } R=0$.

But $P=Q=1$ means both switches are up and nowhere in the problem is mentioned the state of light when both switches are up?

**Reply by spamegg1** [reply]
It says

> Moving either switch to the opposite position turns the lights off if they are on and on if they are off.

- Initially P=0 Q=0 R=0
- If we switch any, lights turn on:
  - P=1 Q=0 R=1
  - P=0 Q=1 R=1
- If we switch the other, then lights turn back off:
  - P=1 Q=1 R=0
  - P=1 Q=1 R=0

You can think of it as a tree if you want
<img width="1170" height="491" alt="Captura de pantalla de 2026-07-10 14-30-34" src="https://github.com/user-attachments/assets/2c1b14dd-a932-47c1-a5ee-d505fdfaf0b9" />

**Reply by mohammadazeemwani** [reply]
okay! got it 🫡
Thank you…(´人｀●)SOOOOO━━＼(´∀｀●)／━━MUCH!!

**Comment by mohammadazeemwani** [comment]
In 3.1.19

<img width="600" alt="image" src="https://github.com/user-attachments/assets/ed3268fa-9c04-4532-bb19-fc115ef29343" />

For parts (b), (d), and (e), which are equivalent to given statement in question, you are saying the same thing as for parts (a), (c).

<p align="center">
$\Large \text{i.e.:  } \forall \text{ integer } n \text{, } n^2 \text{ is even and } n \text{ is even}$
</p>



Why is that?

**Reply by spamegg1** [reply]
Just typo :smile:

**Reply by spamegg1** [reply]
Should be fixed now.
